---
title: Output
layout: default
---

<h1 id="output">Output</h1>
<div class="contrib"><a href="{{ 'resources/pangolin.html' | absolute_url }}">Back to pangolin documentation home page.</a></div>

<br>

<p>pangolin outputs a csv file with taxon name and lineage assigned, one line corresponding to each sequence in the fasta file provided. The following descriptions relate to pangolin 4.0 onwards.</p>

<h3 id="csv-column-headers">CSV column headers</h3>

<div class="contrib">
  <h5 id="taxon">taxon</h5>
  <p>The name of an input query sequence.</p>
</div>

<div class="contrib">
  <h5 id="lineage">lineage</h5>
  <p>The most likely lineage assigned to a given sequence based on the inference engine used and the SARS-CoV-2 diversity <a href="https://github.com/cov-lineages/pango-designation">designated</a>. This assignment may be is sensitive to missing data at key sites.</p>
</div>

<div class="contrib">
  <h5 id="conflict">conflict</h5>
  <p>In the pangoLEARN model, a given sequence gets assigned to the most likely category based on known diversity. If a sequence can fit into more than one category, the conflict score will be greater than 0 and reflect the number of categories the sequence could fit into. If the conflict score is 0, this means that within the current decision tree there is only one category that the sequence could be assigned to.</p>
</div>

<div class="contrib">
  <h5 id="ambiguity_score">ambiguity_score</h5>
  <p>This score is a function of the quantity of missing data in a sequence. It represents the proportion of relevant sites in a sequnece which were imputed to the reference values. A score of 1 indicates that no sites were imputed, while a score of 0 indicates that more sites were imputed than were not imputed. This score only includes sites which are used by the decision tree to classify a sequence.</p>
</div>

<div class="contrib">
  <h5 id="scorpio_call">scorpio_call</h5>
  <p>If a query is assigned a constellation by scorpio this call is output in this column. The full set of constellations searched by default can be found at <a href="https://github.com/cov-lineages/constellations">the constellations repository</a>.</p>
</div>

<div class="contrib">
  <h5 id="scorpio_support">scorpio_support</h5>
  <p>The support score is the proportion of defining variants which have the alternative allele in the sequence.</p>
</div>

<div class="contrib">
  <h5 id="scorpio_conflict">scorpio_conflict</h5>
  <p>The conflict score is the proportion of defining variants which have the reference allele in the sequence. Ambiguous/other non-ref/alt bases at each of the variant positions contribute only to the denominators of these scores</p>
</div>

<div class="contrib">
  <h5 id="scorpio_notes">scorpio_notes</h5>
  <p>A notes column specific to the scorpio output.</p>
</div>

<div class="contrib">
  <h5 id="version">version</h5>
  <p>A version number that represents both pangolin-data version number, which as of pangolin 4.0 corresponds to the pango-designation version used to prepare the inference files. For example:</p>
  <ul>
    <li>PANGO-1.2 indicates an identical sequence has been previously designated this lineage, and has so gone through manual curation. The number 1.2 indicates the version of <a href="https://github.com/cov-lineages/pango-designation">pango-designation</a> that this assignment is based on. These hashes and pango-designation version is bundled with the pangoLEARN and UShER models.</li>
    <li>PLEARN-1.2 indicates that this sequence is different from any previously designated and that the pangoLEARN model was used as an inference engine to predict the most likely lineage based on the given version of <a href="https://github.com/cov-lineages/pango-designation">pango-designation</a> upon which the pangoLEARN model was trained.</li>
    <li>PUSHER-1.2 indicates that this sequence is different from any previously designated and that UShER was used as an inference engine with fast tree placement and parsimony-based lineage assignment, based on a guide tree (protobuf) file built from the data in a given <a href="https://github.com/cov-lineages/pango-designation">pango-designation</a> release version.</li>
  </ul>
</div>

<div class="contrib">
  <h5 id="pangolin_version">pangolin_version</h5>
  <p>The version of <a href="https://github.com/cov-lineages/pangolin">pangolin</a> software running.</p>
</div>

<div class="contrib">
  <h5 id="scorpio_version">scorpio_version</h5>
  <p>The version of the <a href="https://github.com/cov-lineages/scorpio">scorpio</a> software installed.</p>
</div>

<div class="contrib">
  <h5 id="constellation_version">constellation_version</h5>
  <p>The version of <a href="https://github.com/cov-lineages/constellations">constellations</a> that scorpio has used to curate the lineage assignment.</p>
</div>

<div class="contrib">
  <h5 id="is_designated">is_designated</h5>
  <p>A boolean (True/False) column indicating whether that particular sequence has been offically designated a lineage.</p>
</div>

<div class="contrib">
  <h5 id="qc_status">qc_status</h5>
  <p>Indicates whether the sequence passed the QC thresholds for minimum length and maximum N content.</p>
</div>

<div class="contrib">
  <h5 id="qc_notes">qc_notes</h5>
  <p>Notes specific to the QC checks run on the sequences.</p>
</div>

<div class="contrib">
  <h5 id="note">note</h5>
  <p>If any conflicts from the decision tree, this field will output the alternative assignments. If the sequence failed QC this field will describe why. If the sequence met the SNP thresholds for scorpio to call a constellation, itâ€™ll describe the exact SNP counts of Alt, Ref and Amb (Alternative, reference and ambiguous) alleles for that call.</p>
</div>
<br>
<h3 id="example-output">Example output</h3><br>
<table id="myTable" class="table tightTable">
  <thead class="thead-light">
    <tr>
      <th>taxon</th>
      <th>lineage</th>
      <th>conflict</th>
      <th>ambiguity_<br />score</th>
      <th>scorpio_<br />call</th>
      <th>scorpio_<br />support</th>
      <th>scorpio_<br />conflict</th>
      <th>scorpio_<br />notes</th>
      <th>version</th>
      <th>pangolin_<br />version</th>
      <th>scorpio_<br />version</th>
      <th>constellation<br>_version</th>
      <th>is_designated</th>
      <th>qc_status</th>
      <th>qc_notes</th>
      <th>note</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Virus1</td>
      <td>B.1.617.1</td>
      <td>0.0</td>
      <td></td>
      <td>B.1.617.1-like</td>
      <td>1.0</td>
      <td>0.0</td>
      <td>scorpio call: Alt alleles 11; Ref alleles 0; Amb alleles 0; Oth alleles 0</td>
      <td>PUSHER-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>False</td>
      <td>pass</td>
      <td>Ambiguous_content:0.02</td>
      <td></td>
    </tr>
    <tr>
      <td>Virus2</td>
      <td>B.1.1.7</td>
      <td>0.0</td>
      <td></td>
      <td>Alpha (B.1.1.7-like)</td>
      <td>0.91</td>
      <td>0.04</td>
      <td>scorpio call: Alt alleles 21; Ref alleles 1; Amb alleles 1; Oth alleles 0</td>
      <td>PUSHER-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>False</td>
      <td>pass</td>
      <td>Ambiguous_content:0.02</td>
      <td></td>
    </tr>
    <tr>
      <td>Virus3</td>
      <td>A</td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td>PUSHER-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>False</td>
      <td>pass</td>
      <td>Ambiguous_content:0.02</td>
      <td></td>
    </tr>
    <tr>
      <td>Virus4</td>
      <td>B</td>
      <td>0.5</td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td>PANGO-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>True</td>
      <td>pass</td>
      <td>Ambiguous_content:0.02</td>
      <td>Assigned from designation hash.</td>
    </tr>
    <tr>
      <td>Virus5</td>
      <td>B.1.314</td>
      <td>0.0</td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td>PANGO-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>True</td>
      <td>pass</td>
      <td>Ambiguous_content:0.02</td>
      <td>Assigned from designation hash.</td>
    </tr>
    <tr>
      <td>Virus6</td>
      <td>None</td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td>PUSHER-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>False</td>
      <td>fail</td>
      <td>Ambiguous_content:0.9</td>
      <td></td>
    </tr>
    <tr>
      <td>Virus7</td>
      <td>None</td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td>PUSHER-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>False</td>
      <td>fail</td>
      <td>Ambiguous_content:0.98</td>
      <td></td>
    </tr>
    <tr>
      <td>Virus8</td>
      <td>None</td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td></td>
      <td>PUSHER-1.2.101</td>
      <td>4.0</td>
      <td>0.3.16</td>
      <td>v0.1.3</td>
      <td>False</td>
      <td>fail</td>
      <td></td>
      <td>Failed to map</td>
    </tr>
  </tbody>
</table>
