<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">@import url('https://themes.googleusercontent.com/fonts/css?kit=U_HdmvgDLxl92Im33obxxI55b3GVhHISOSUe9YJyFq1U7EveXFbMk4b4K8pExx8kbWD1fmVJrF79R3SeirZgS1RtI6M9mJ6NLMdQfNgV2chCYWjk0ez-NKkYqBB1SMeH');ul.lst-kix_j2cbuqlpa9ij-6{list-style-type:none}ul.lst-kix_j2cbuqlpa9ij-5{list-style-type:none}ul.lst-kix_j2cbuqlpa9ij-8{list-style-type:none}ul.lst-kix_j2cbuqlpa9ij-7{list-style-type:none}.lst-kix_fyfag74htt8u-3>li:before{content:"\0025cf  "}.lst-kix_fyfag74htt8u-5>li:before{content:"\0025a0  "}ul.lst-kix_j2cbuqlpa9ij-2{list-style-type:none}ul.lst-kix_j2cbuqlpa9ij-1{list-style-type:none}ul.lst-kix_j2cbuqlpa9ij-4{list-style-type:none}ul.lst-kix_j2cbuqlpa9ij-3{list-style-type:none}.lst-kix_fyfag74htt8u-0>li:before{content:"\0025cf  "}.lst-kix_fyfag74htt8u-4>li:before{content:"\0025cb  "}.lst-kix_y3dse7uglisi-8>li:before{content:"\0025a0  "}.lst-kix_fyfag74htt8u-7>li:before{content:"\0025cb  "}.lst-kix_fyfag74htt8u-6>li:before{content:"\0025cf  "}.lst-kix_y3dse7uglisi-3>li:before{content:"\0025cf  "}.lst-kix_y3dse7uglisi-4>li:before{content:"\0025cb  "}.lst-kix_fyfag74htt8u-8>li:before{content:"\0025a0  "}.lst-kix_y3dse7uglisi-7>li:before{content:"\0025cb  "}.lst-kix_y3dse7uglisi-5>li:before{content:"\0025a0  "}.lst-kix_y3dse7uglisi-6>li:before{content:"\0025cf  "}ul.lst-kix_fyfag74htt8u-3{list-style-type:none}ul.lst-kix_fyfag74htt8u-2{list-style-type:none}ul.lst-kix_fyfag74htt8u-5{list-style-type:none}ul.lst-kix_fyfag74htt8u-4{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-4.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-4 0}ul.lst-kix_fyfag74htt8u-7{list-style-type:none}ul.lst-kix_fyfag74htt8u-6{list-style-type:none}ul.lst-kix_fyfag74htt8u-8{list-style-type:none}.lst-kix_sj5mk9oydhv0-3>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-3}.lst-kix_y3dse7uglisi-0>li:before{content:"\0025cf  "}.lst-kix_y3dse7uglisi-1>li:before{content:"\0025cb  "}.lst-kix_y3dse7uglisi-2>li:before{content:"\0025a0  "}.lst-kix_sj5mk9oydhv0-5>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-5}.lst-kix_9arkyp9zvjki-4>li:before{content:"\0025cb  "}.lst-kix_9arkyp9zvjki-3>li:before{content:"\0025cf  "}.lst-kix_ik5tnb9cepy4-8>li:before{content:"-  "}.lst-kix_9arkyp9zvjki-1>li:before{content:"\0025cb  "}.lst-kix_ik5tnb9cepy4-6>li:before{content:"-  "}.lst-kix_ik5tnb9cepy4-7>li:before{content:"-  "}.lst-kix_9arkyp9zvjki-0>li:before{content:"\0025cf  "}.lst-kix_9arkyp9zvjki-2>li:before{content:"\0025a0  "}.lst-kix_fyfag74htt8u-1>li:before{content:"\0025cb  "}ul.lst-kix_fyfag74htt8u-1{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-7.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-7 0}ul.lst-kix_fyfag74htt8u-0{list-style-type:none}.lst-kix_fyfag74htt8u-2>li:before{content:"\0025a0  "}.lst-kix_ik5tnb9cepy4-2>li:before{content:"-  "}.lst-kix_ik5tnb9cepy4-3>li:before{content:"-  "}.lst-kix_ik5tnb9cepy4-0>li:before{content:"-  "}.lst-kix_ik5tnb9cepy4-1>li:before{content:"-  "}.lst-kix_ik5tnb9cepy4-4>li:before{content:"-  "}.lst-kix_ik5tnb9cepy4-5>li:before{content:"-  "}ul.lst-kix_9arkyp9zvjki-0{list-style-type:none}ul.lst-kix_9arkyp9zvjki-1{list-style-type:none}ul.lst-kix_9arkyp9zvjki-2{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-5.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-5 0}ul.lst-kix_9arkyp9zvjki-3{list-style-type:none}ul.lst-kix_9arkyp9zvjki-4{list-style-type:none}ul.lst-kix_9arkyp9zvjki-5{list-style-type:none}ul.lst-kix_9arkyp9zvjki-6{list-style-type:none}ul.lst-kix_9arkyp9zvjki-7{list-style-type:none}ul.lst-kix_9arkyp9zvjki-8{list-style-type:none}.lst-kix_sj5mk9oydhv0-2>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-2}.lst-kix_sj5mk9oydhv0-6>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-6}.lst-kix_j2cbuqlpa9ij-0>li:before{content:"\0025cf  "}.lst-kix_j2cbuqlpa9ij-1>li:before{content:"\0025cb  "}ul.lst-kix_dl2af9g3g50m-8{list-style-type:none}ul.lst-kix_dl2af9g3g50m-7{list-style-type:none}ul.lst-kix_dl2af9g3g50m-6{list-style-type:none}ul.lst-kix_dl2af9g3g50m-5{list-style-type:none}ul.lst-kix_dl2af9g3g50m-4{list-style-type:none}ul.lst-kix_dl2af9g3g50m-3{list-style-type:none}ul.lst-kix_dl2af9g3g50m-2{list-style-type:none}ul.lst-kix_dl2af9g3g50m-1{list-style-type:none}ul.lst-kix_dl2af9g3g50m-0{list-style-type:none}.lst-kix_j2cbuqlpa9ij-7>li:before{content:"\0025cb  "}.lst-kix_j2cbuqlpa9ij-8>li:before{content:"\0025a0  "}ol.lst-kix_sj5mk9oydhv0-0.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-0 0}.lst-kix_j2cbuqlpa9ij-2>li:before{content:"\0025a0  "}.lst-kix_j2cbuqlpa9ij-4>li:before{content:"\0025cb  "}.lst-kix_j2cbuqlpa9ij-3>li:before{content:"\0025cf  "}ol.lst-kix_sj5mk9oydhv0-6.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-6 0}.lst-kix_j2cbuqlpa9ij-6>li:before{content:"\0025cf  "}ol.lst-kix_sj5mk9oydhv0-3.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-3 0}.lst-kix_j2cbuqlpa9ij-5>li:before{content:"\0025a0  "}ul.lst-kix_ik5tnb9cepy4-8{list-style-type:none}.lst-kix_k1z0w9xyijrs-3>li:before{content:"\0025cf  "}.lst-kix_dl2af9g3g50m-6>li:before{content:"\0025cf  "}.lst-kix_dl2af9g3g50m-7>li:before{content:"\0025cb  "}.lst-kix_k1z0w9xyijrs-2>li:before{content:"\0025a0  "}.lst-kix_k1z0w9xyijrs-4>li:before{content:"\0025cb  "}ul.lst-kix_ik5tnb9cepy4-4{list-style-type:none}.lst-kix_dl2af9g3g50m-4>li:before{content:"\0025cb  "}.lst-kix_dl2af9g3g50m-5>li:before{content:"\0025a0  "}.lst-kix_dl2af9g3g50m-8>li:before{content:"\0025a0  "}ul.lst-kix_ik5tnb9cepy4-5{list-style-type:none}ul.lst-kix_ik5tnb9cepy4-6{list-style-type:none}ul.lst-kix_ik5tnb9cepy4-7{list-style-type:none}ul.lst-kix_ik5tnb9cepy4-0{list-style-type:none}ul.lst-kix_ik5tnb9cepy4-1{list-style-type:none}ul.lst-kix_ik5tnb9cepy4-2{list-style-type:none}.lst-kix_k1z0w9xyijrs-6>li:before{content:"\0025cf  "}ul.lst-kix_ik5tnb9cepy4-3{list-style-type:none}.lst-kix_sj5mk9oydhv0-0>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-0}.lst-kix_k1z0w9xyijrs-5>li:before{content:"\0025a0  "}.lst-kix_sj5mk9oydhv0-8>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-8}.lst-kix_dl2af9g3g50m-0>li:before{content:"\0025cf  "}.lst-kix_dl2af9g3g50m-1>li:before{content:"\0025cb  "}.lst-kix_dl2af9g3g50m-2>li:before{content:"\0025a0  "}.lst-kix_dl2af9g3g50m-3>li:before{content:"\0025cf  "}.lst-kix_k1z0w9xyijrs-0>li:before{content:"\0025cf  "}.lst-kix_k1z0w9xyijrs-1>li:before{content:"\0025cb  "}ol.lst-kix_sj5mk9oydhv0-2{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-3{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-8{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-4{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-5{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-6{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-7{list-style-type:none}ol.lst-kix_sj5mk9oydhv0-8{list-style-type:none}.lst-kix_sj5mk9oydhv0-7>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-7}ul.lst-kix_k1z0w9xyijrs-1{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-0{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-3{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-2{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-5{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-4{list-style-type:none}ul.lst-kix_y3dse7uglisi-0{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-7{list-style-type:none}ul.lst-kix_k1z0w9xyijrs-6{list-style-type:none}.lst-kix_9arkyp9zvjki-5>li:before{content:"\0025a0  "}.lst-kix_sj5mk9oydhv0-8>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-8,lower-roman) ". "}ol.lst-kix_sj5mk9oydhv0-1.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-1 0}.lst-kix_9arkyp9zvjki-6>li:before{content:"\0025cf  "}.lst-kix_9arkyp9zvjki-7>li:before{content:"\0025cb  "}ol.lst-kix_sj5mk9oydhv0-8.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-8 0}ul.lst-kix_uqgnyuie10hs-2{list-style-type:none}ul.lst-kix_uqgnyuie10hs-3{list-style-type:none}.lst-kix_sj5mk9oydhv0-7>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-7,lower-latin) ". "}ul.lst-kix_uqgnyuie10hs-0{list-style-type:none}ul.lst-kix_uqgnyuie10hs-1{list-style-type:none}.lst-kix_sj5mk9oydhv0-4>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-4,lower-latin) ". "}ul.lst-kix_uqgnyuie10hs-6{list-style-type:none}.lst-kix_k1z0w9xyijrs-7>li:before{content:"\0025cb  "}ul.lst-kix_uqgnyuie10hs-7{list-style-type:none}.lst-kix_9arkyp9zvjki-8>li:before{content:"\0025a0  "}ul.lst-kix_uqgnyuie10hs-4{list-style-type:none}.lst-kix_k1z0w9xyijrs-8>li:before{content:"\0025a0  "}ul.lst-kix_uqgnyuie10hs-5{list-style-type:none}.lst-kix_sj5mk9oydhv0-6>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-6,decimal) ". "}.lst-kix_sj5mk9oydhv0-5>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-5,lower-roman) ". "}ul.lst-kix_uqgnyuie10hs-8{list-style-type:none}.lst-kix_sj5mk9oydhv0-0>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-0,decimal) ". "}.lst-kix_sj5mk9oydhv0-3>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-3,decimal) ". "}.lst-kix_uqgnyuie10hs-8>li:before{content:"\0025a0  "}.lst-kix_uqgnyuie10hs-5>li:before{content:"\0025a0  "}.lst-kix_uqgnyuie10hs-7>li:before{content:"\0025cb  "}.lst-kix_sj5mk9oydhv0-2>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-2,lower-roman) ". "}.lst-kix_sj5mk9oydhv0-1>li:before{content:"" counter(lst-ctn-kix_sj5mk9oydhv0-1,lower-latin) ". "}.lst-kix_uqgnyuie10hs-6>li:before{content:"\0025cf  "}ul.lst-kix_y3dse7uglisi-2{list-style-type:none}ul.lst-kix_y3dse7uglisi-1{list-style-type:none}ul.lst-kix_y3dse7uglisi-4{list-style-type:none}ul.lst-kix_y3dse7uglisi-3{list-style-type:none}.lst-kix_uqgnyuie10hs-1>li:before{content:"\0025cb  "}.lst-kix_uqgnyuie10hs-3>li:before{content:"\0025cf  "}ul.lst-kix_y3dse7uglisi-6{list-style-type:none}ul.lst-kix_y3dse7uglisi-5{list-style-type:none}ul.lst-kix_y3dse7uglisi-8{list-style-type:none}ul.lst-kix_y3dse7uglisi-7{list-style-type:none}.lst-kix_uqgnyuie10hs-0>li:before{content:"\0025cf  "}.lst-kix_uqgnyuie10hs-4>li:before{content:"\0025cb  "}ul.lst-kix_j2cbuqlpa9ij-0{list-style-type:none}.lst-kix_sj5mk9oydhv0-4>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-4}ol.lst-kix_sj5mk9oydhv0-2.start{counter-reset:lst-ctn-kix_sj5mk9oydhv0-2 0}ol.lst-kix_sj5mk9oydhv0-0{list-style-type:none}.lst-kix_sj5mk9oydhv0-1>li{counter-increment:lst-ctn-kix_sj5mk9oydhv0-1}ol.lst-kix_sj5mk9oydhv0-1{list-style-type:none}.lst-kix_uqgnyuie10hs-2>li:before{content:"\0025a0  "}ol{margin:0;padding:0}table td,table th{padding:0}.c7{border-right-style:solid;padding-top:5pt;border-top-width:0pt;border-right-width:0pt;padding-left:5pt;padding-bottom:5pt;line-height:1.0;border-left-width:0pt;border-top-style:solid;background-color:#404446;border-left-style:solid;border-bottom-width:0pt;border-bottom-style:solid;orphans:2;widows:2;text-align:left;padding-right:5pt}.c4{color:#000000;font-weight:700;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Helvetica Neue";font-style:normal}.c9{color:#000000;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Courier New";font-style:normal}.c1{color:#f3f3f3;font-weight:400;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Consolas";font-style:normal}.c0{color:#000000;font-weight:300;text-decoration:none;vertical-align:baseline;font-size:12pt;font-family:"Helvetica Neue";font-style:normal}.c14{color:#000000;font-weight:300;text-decoration:none;vertical-align:baseline;font-size:24pt;font-family:"Helvetica Neue";font-style:normal}.c18{padding-top:24pt;padding-bottom:6pt;line-height:1.1500000000000001;orphans:2;widows:2;text-align:left}.c3{padding-top:0pt;padding-bottom:0pt;line-height:1.0;orphans:2;widows:2;text-align:left}.c19{color:#000000;text-decoration:none;vertical-align:baseline;font-size:16pt;font-style:normal}.c8{color:#f3f3f3;text-decoration:none;vertical-align:baseline;font-size:13pt;font-style:normal}.c17{color:#f3f3f3;text-decoration:none;vertical-align:baseline;font-size:11pt;font-style:normal}.c10{text-decoration-skip-ink:none;-webkit-text-decoration-skip:none;color:#1155cc;text-decoration:underline}.c11{background-color:#ffffff;max-width:451.4pt;padding:72pt 72pt 72pt 72pt}.c13{background-color:#404446;color:#f3f3f3}.c12{color:inherit;text-decoration:inherit}.c2{font-weight:400;font-family:"Consolas"}.c6{font-weight:700;font-family:"Helvetica Neue"}.c16{font-weight:400;font-family:"Courier New"}.c5{height:12pt}.c15{font-size:13pt}.title{padding-top:24pt;color:#000000;font-weight:300;font-size:24pt;padding-bottom:6pt;font-family:"Helvetica Neue";line-height:1.1500000000000001;orphans:2;widows:2;text-align:left}.subtitle{padding-top:18pt;color:#666666;font-weight:300;font-size:14pt;padding-bottom:4pt;font-family:"Helvetica Neue";line-height:1.1500000000000001;font-style:italic;orphans:2;widows:2;text-align:left}li{color:#000000;font-weight:300;font-size:12pt;font-family:"Helvetica Neue"}p{margin:0;color:#000000;font-weight:300;font-size:12pt;font-family:"Helvetica Neue"}h1{padding-top:0pt;color:#000000;font-weight:700;font-size:14pt;padding-bottom:0pt;font-family:"Helvetica Neue";line-height:1.1500000000000001;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-weight:300;font-size:14pt;padding-bottom:4pt;font-family:"Helvetica Neue";line-height:1.1500000000000001;orphans:2;widows:2;text-align:left}h3{padding-top:14pt;color:#666666;font-weight:700;font-size:12pt;padding-bottom:4pt;font-family:"Helvetica Neue";line-height:1.1500000000000001;orphans:2;widows:2;text-align:left}h4{border-right-style:solid;color:#f3f3f3;border-top-width:0pt;font-weight:300;padding-left:5pt;font-size:12pt;background-color:#404446;border-left-style:solid;font-family:"Consolas";orphans:2;text-align:left;padding-top:5pt;border-right-width:0pt;padding-bottom:5pt;line-height:1.0;border-left-width:0pt;border-top-style:solid;border-bottom-width:0pt;border-bottom-style:solid;widows:2;padding-right:5pt}h5{padding-top:11pt;color:#666666;font-weight:700;font-size:10pt;padding-bottom:2pt;font-family:"Cabin";line-height:1.1500000000000001;orphans:2;widows:2;text-align:left}h6{padding-top:10pt;color:#666666;font-weight:300;font-size:10pt;padding-bottom:2pt;font-family:"Cabin";line-height:1.1500000000000001;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c11 doc-content"><p class="c18 title" id="h.exnwvbwzimjq"><span><a id="installation">Installing and running </span><span class="c2">civet</a></span><span class="c14">&nbsp;on the command line</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c4">IS THIS YOUR FIRST TIME RUNNING CIVET? IF NO, SKIP TO THE NEXT SECTION.</span></p><p class="c3 c5"><span class="c0"></span></p><a id="id.1i0tc2j350wx"></a><p class="c3"><span class="c4">INSTALLATION</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">If this is your first time running civet, you&rsquo;ll need to clone the repository and install it. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">I like to keep my repositories in the same place so they&rsquo;re easy to find, so below I&rsquo;m making a directory in my home directory (~) called repositories and moving into it with cd. </span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.6pdd4qiwhxh9"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;cd ~</span></h4><h4 class="c7" id="h.g41zek6e8v0r"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;mkdir repositories</span></h4><h4 class="c7" id="h.qbqfgxe44y2z"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;cd repositories</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Now we&rsquo;re ready to clone the civet github repository. This creates a local copy of the civet repository on your computer. </span></p><h4 class="c7" id="h.cttowvl1x7eu"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;git clone https://github.com/artic-network/civet.git</span></h4><h4 class="c7" id="h.dwfgwvjz4tjr"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;cd civet</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">This directory remains linked to the original github copy, so if you need to update it or get any changes you can do that from this location with the following command:</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.2xvtgqo2r2qn"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;git pull</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">We now need to create the civet environment. Hopefully we have mamba installed, check if you do with the following command:</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.954sd92cqzmn"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;mamba &ndash;version</span></h4><p class="c3"><span class="c0">If you see the message &ldquo;mamba not found&rdquo; then you should install mamba with:</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.w7mxo84yetvd"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;conda install mamba -c conda-forge</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">If you saw the version of mamba printed, you can now create your civet environment.</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.vfpz8nw3xel3"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;mamba env create -f environment.yml</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Activate the civet conda environment:</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.5dfra4xy7pp4"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;conda activate civet</span></h4><h4 class="c7" id="h.1w5769hpigkv"><span class="c1">(civet) artic$ &nbsp; &nbsp;</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span>The &lsquo;</span><span class="c2 c13">(civet)</span><span>&rsquo; tells you that the </span><span class="c2">civet</span><span class="c0">&nbsp;environment is activated.</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Now you&rsquo;ll need run install the civet python package while you&rsquo;re in the environment:</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.7u1nvmyb8jkf"><span class="c1">(civet) artic$ pip install . &nbsp; </span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Don&rsquo;t forget the &lsquo;.&rsquo; in the above command, it indicates you want to install the package in the current directory.</span></p><p class="c3 c5"><span class="c0"></span></p><a id="id.flsri8d6y9m"></a><p class="c3"><span class="c4"><a id="check_the_install_worked">CHECK THE INSTALL WORKED</a></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Now you will have civet installed, check this worked with:</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.rgy049tq5qhf"><span class="c1">(civet) artic$ civet --version </span></h4><h4 class="c7" id="h.oize35b2jp7h"><span class="c1">(civet) artic$ civet --art </span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">If no errors have come up yet, you should now be ready to run civet! </span></p><p class="c3 c5"><span class="c0"></span></p><a id="id.rgyqicjwtbzd"></a><p class="c3"><span class="c6"><a id="updating">UPDATING CIVET</a></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Double check you&rsquo;re in the civet conda environment:</span></p><h4 class="c7" id="h.2draivsh0dkb"><span>(base) </span><span>artic$</span><span class="c1">&nbsp;conda activate civet</span></h4><h4 class="c7" id="h.ejf804lcxjjg"><span class="c1">(civet) artic$ civet --help &nbsp; &nbsp;</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Good practice is to check for and install any updates:</span></p><h4 class="c7" id="h.easw03uqwwxp"><span>(civet) </span><span>artic$</span><span class="c1">&nbsp;cd ~/repositories/civet</span></h4><h4 class="c7" id="h.2ykggvmlk2ow"><span class="c1">(civet) artic$ git pull </span></h4><h4 class="c7" id="h.hiknm94ub7zi"><span>(civet) artic$ pip install . </span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span>Once we know we have civet installed, you want to first prepare a </span><span class="c6">background dataset</span><span class="c0">. This dataset should consist of an aligned fasta file, a metadata file that contains a row of information for each sequence record in the fasta file, and optionally a SNP file. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><a id="id.g4vjy5e7zbpg"></a><p class="c3"><span class="c4"><a id="background_dataset">GETTING A GOOD BACKGROUND DATASET</a></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">In an ideal world we would use all sequences (&gt;12 million) on GISAID, but this is untenable on local laptop computers. In this case we&rsquo;ll want to tailor-make a background dataset, to cross-reference using civet.</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Minimally</span><span class="c0">, you should have:</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">1. FASTA </span><span class="c0">(an aligned fasta file)</span></p><p class="c3"><span class="c0">&gt;seq1</span></p><p class="c3"><span class="c0">ACAGTCGCT&hellip;.CGATCATACGA</span></p><p class="c3"><span class="c0">&gt;seq2</span></p><p class="c3"><span class="c0">ACA--CGCT&hellip;.CGACCGTACGA</span></p><p class="c3"><span class="c0">&gt;se3</span></p><p class="c3"><span class="c0">ACAGTCGCT&hellip;.CGATCGTACGA</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">2. METADATA </span><span class="c0">(minimally a column that matches the sequence name, and any other information you might have about the sequences)</span></p><p class="c3"><span class="c0">name,sample_date,lineage, &hellip;</span></p><p class="c3"><span class="c0">seq2,2022-06-02,BA.5, &hellip;</span></p><p class="c3"><span class="c0">seq2,2022-06-01,BA.4, &hellip;</span></p><p class="c3"><span class="c0">seq3,2022-06-04,BA.4.1, &hellip;</span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c4">Example background dataset</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">A good place to start getting a background dataset is on gisaid.org. The following is a step-by-step guide to downloading the latest next-strain background sample set (~3K sequences). </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 601.70px; height: 365.33px;"><img alt="" src="images/image3.png" style="width: 601.70px; height: 365.33px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c0">Step 1: Log into gisaid (or register for an account if you don&rsquo;t already have one) and navigate to the downloads window on EpiCoV. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 547.50px; height: 409.92px;"><img alt="" src="images/image1.png" style="width: 547.50px; height: 409.92px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c0">Step 2: Scroll down to the bottom</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 554.90px; height: 346.50px;"><img alt="" src="images/image2.png" style="width: 554.90px; height: 346.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c0">Step 3: Click on the nextregions download option</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 558.24px; height: 359.41px;"><img alt="" src="images/image5.png" style="width: 558.24px; height: 359.41px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c0">Step 4: Download one of the latest downsamples, for example the Africa build which is enriched for sequences from the African continent. Which one you download will depend on what you want to investigate, where you are based and what locations are most connected to your location of interest. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 571.26px; height: 308.50px;"><img alt="" src="images/image6.png" style="width: 571.26px; height: 308.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c0">Step 5: Unzip the downloaded file and look in the directory. You will see 4 files. One is the GISAID user agreement (readme.txt), one is a json file with the format of data for running nextstrain auspice, but the two we&rsquo;re interested in using with civet are the fasta file (the sequences) and the tsv file (the metadata).</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Step 6: We now need to curate this metadata to make sure our sequences can be matched with the metadata file (this mostly involves removing spaces from the sequence file name). By default the nextstrain tsv file has a column called &lsquo;strain&rsquo; that should match the sequence file, this is not always the case so with civet we will create a new file with a modified column that will successfully match. We also need to make sure the file is aligned in a way consistent with the way civet will align new sequences. Both of these steps are run with the following command. Note, the backslashes indicate the command continues onto the next line.</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.8s3jq0de7heu"><span>(civet) artic$ </span><span class="c8 c2">civet -bd align_curate \</span></h4><h4 class="c7" id="h.191bt1d920ez"><span class="c8 c2">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; -bd-seqs hcov_africa.fasta \</span></h4><h4 class="c7" id="h.knj6lu6o5dtj"><span class="c2 c8">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; -bd-metadata hcov_africa.tsv \</span></h4><h4 class="c7" id="h.ewevq1zlucy"><span class="c15">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; --sequence-id-column strain</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">This command produces a directory called civet_data, that contains the following files:</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 500.50px; height: 102.26px;"><img alt="" src="images/image4.png" style="width: 500.50px; height: 102.26px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c4">What&rsquo;s changed in the metadata file:</span></p><p class="c3"><span class="c0">The new column that will match your sequences to the metadata will be `modified_{orginal_name}`, i.e. because `strain` is what nextstrain call the column, by default we will now have a column called `modifed_strain` that we will use to match. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span>You can rename this column to something more descriptive like &ldquo;sequence_name&rdquo;, which is what civet looks for by default, or just make sure to supply `</span><span class="c16">--background-id-column modified_strain</span><span class="c0">` when running civet. </span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c4">Tell civet where this data is:</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span>When running civet, you can point the software to this directory with `</span><span class="c16">-d/&ndash;datadir</span><span class="c0">`, or set it as a command line variable with:</span></p><h4 class="c7" id="h.j43d4zk7ozkg"><span class="c1">(civet) artic$ CIVET_DATA=/mnt/c/Users/artic/&hellip;/path/to/civet_data</span></h4><h4 class="c7" id="h.4gstrumcxqzs"><span class="c1">(civet) artic$ echo $CIVET_DATA &nbsp; &nbsp;</span></h4><p class="c3"><span class="c0">You can even add this variable to your bash_profile, or zshrc so that the path is always loaded by default. If updating this data, by saving it in the same directory and overwriting with latest data you can maintain an updated background dataset.</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Bear in mind that you will only want to have one fasta file, one metadata file and one SNP file in this directory though, or else you&rsquo;ll need to explicitly tell civet which ones to use. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c4">ALTERNATIVE BACKGROUND DATASETS</span></p><p class="c3"><span class="c0">This is just one way of accessing a background dataset. You can create a variety of custom background datasets by accessing gisaid or ncbi SARS-CoV-2 genome data. You might also want to create a local dataset with all the previously sequenced data from your institute, so that when you have new sequences you can compare them against all the samples you&rsquo;ve previously sequenced.</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><a id="id.ajvw8960pz05"></a><p class="c3"><span class="c4">FINALLY we&rsquo;ve gotten to the point where we can begin to run civet</span></p><p class="c3 c5"><span class="c4"></span></p><p class="c3"><span class="c0">Civet can accept information in two ways: 1. From the command line arguments or 2. From a configuration file in yaml format. If you supply a combination of the two (some on the command line, some from a config file) the command line arguments will always take priority and overrule the config option. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6 c19"><a id="example_use_cases">Example use cases:</a></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Scenario</span><span class="c0">: One of the simplest use cases, you want to summarise all the data from a given country in your background data.</span></p><p class="c3"><span class="c6">Input data: </span><span class="c0">The background dataset</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.jm1jsf6420k5"><span class="c1">(civet) artic$ civet -d path/to/civet_data \</span></h4><h4 class="c7" id="h.39kfn0fl41ot"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-fm country=Ghana \</span></h4><h4 class="c7" id="h.ublsgrvk3843"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-bicol modified_strain </span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">This will produce a report that highlights all sequences from Ghana in your background dataset as sequences of interest. </span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Scenario</span><span class="c0">: If we want to filter this further, we might just want to summarise all the data between April and July 2022 from Ghana in your background data.</span></p><p class="c3"><span class="c6">Input data: </span><span class="c0">The background dataset</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.tgs07471scon"><span class="c1">(civet) artic$ civet -d path/to/civet_data \</span></h4><h4 class="c7" id="h.ubctzn965f6r"><span>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-fm country=Ghana date=2022-04-01:2022-07-01\</span></h4><h4 class="c7" id="h.j8s7i5wqg7n0"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-bicol modified_strain </span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">This also highlights why it&rsquo;s so important to keep consistent with date formats in metadata. We recommend using ISO format YYYY-MM-DD for all dates in your metadata. This command will produce a report that highlights all sequences from Ghana with date in the timeframe specified.</span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">If you have any particular sequences you&rsquo;d like to investigate in your background data you can specify them with the -ids flag:</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Scenario</span><span class="c0">: a small number of sequences of interest in your background dataset</span></p><p class="c3"><span class="c6">Input data: </span><span class="c0">The background dataset</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.pkskxsivs7uz"><span class="c1">(civet) artic$ civet -d path/to/civet_data \</span></h4><h4 class="c7" id="h.yzk9j0h6yyqy"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-ids EPI_ISL_12012922,EPI_ISL_12970473 \</span></h4><h4 class="c7" id="h.quh6lyd3ooeb"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-bicol gisaid_epi_isl</span></h4><h4 class="c7" id="h.4clm3lh6qpfb"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-sicol modified_strain </span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">This will produce a report that highlights the two sequences with GISID IDs EPI_ISL_12012922 and EPI_ISL_12970473 in your background dataset. You need to specify what ID column to match with -bicol (background_id_column) and then still make sure you&rsquo;re matching the sequences with the correct column name (-sicol/&ndash;sequence-id-column).</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Scenario</span><span class="c0">: We have new data in a fasta file that has just been sequenced and we want to produce a summary report of this data.</span></p><p class="c3"><span class="c6">Input data: </span><span class="c0">The background dataset, new fasta file with genome sequences</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.xlg2ogg6dr8c"><span class="c1">(civet) artic$ civet -d path/to/civet_data \</span></h4><h4 class="c7" id="h.983ecp1woopz"><span>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-bicol modified_strain \</span></h4><h4 class="c7" id="h.s4aiebnekn6l"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-f path/to/input_sequences.fasta </span></h4><h4 class="c7" id="h.firdnmqudiz1"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">This will add your new sequences from the fasta file into the report and produce a report that highlights these sequences within the context of the background dataset.</span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Scenario</span><span class="c0">: We have new data in a fasta file AND some associated metadata with epidemiological information, and we want to produce a summary report of this data.</span></p><p class="c3"><span class="c6">Input data: </span><span class="c0">The background dataset, new fasta file with genome sequences, metadata about the new sequences in a csv file.</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.skqb4epicq80"><span class="c1">(civet) artic$ civet -d path/to/civet_data \</span></h4><h4 class="c7" id="h.yeo2zuh4c9ms"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-bicol modified_strain \</span></h4><h4 class="c7" id="h.3fo5tg6zavz8"><span>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-f path/to/new_genome_sequences.fasta \<br> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; -i path/to/new_genome_sequences_metadata.csv</span></h4><h4 class="c7" id="h.u7gu8s1ui0gl"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span></h4><p class="c3"><span class="c0">To match the metadata up with your new genome sequences you need to make sure there is a column in the metadata that corresponds to the name of the sequence in the fasta file. By default civet will look for a column `name`. If your column is called something other than name, you&rsquo;ll need to specify what the column is when running civet. For example, if that column was `sequence_id`:</span></p><p class="c3 c5"><span class="c0"></span></p><h4 class="c7" id="h.o6kq3vbwok98"><span class="c1">(civet) artic$ civet -d path/to/civet_data \</span></h4><h4 class="c7" id="h.3hojrh1mpgra"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-bicol modified_strain \</span></h4><h4 class="c7" id="h.rqsray3ae1sa"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-f path/to/new_genome_sequences.fasta \<br> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; -i path/to/new_genome_sequences_metadata.csv \</span></h4><h4 class="c7" id="h.e4z3h69vaqg5"><span class="c1">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;-icol sequence_id</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">This will add your new sequences from the fasta file into the report and produce a report that highlights these sequences within the context of the background dataset. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Scenario: </span><span class="c0">Using a configuration file</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">As you can see, the commands for civet can get a little unwieldy as more and more options get specified. If you&rsquo;re going to be running similar reports, with similar-structured data, you can create a configuration file that can be used again and again to create a report. The above command can be converted into a configuration file as such:</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c9">config.yaml</span></p><h4 class="c7" id="h.16wuluwzltpq"><span class="c1">datadir: path/to/civet_data </span></h4><h4 class="c7" id="h.pb79pmdnmlei"><span class="c1">background_id_column: modified_strain </span></h4><h4 class="c7" id="h.sqsl88se4mdt"><span class="c1">input_sequences: path/to/new_genome_sequences.fasta <br>input_metadata: path/to/new_genome_sequences_metadata.csv &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; input_id_column: sequence_id</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c6">Note</span><span class="c0">: The configuration file keys are the long-form command line arguments with either dashes or underscores (so input-metadata and input_metdata are both accepted).</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">&nbsp;Now we can continue adding to the configuration to create a bespoke report:</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c9">config.yaml</span></p><h4 class="c7" id="h.z0mme321h5"><span class="c1">datadir: path/to/civet_data </span></h4><h4 class="c7" id="h.f6wkfy7fka4t"><span class="c1">background_id_column: modified_strain <br></span></h4><h4 class="c7" id="h.c5bxgqyjpnxi"><span class="c1">input_sequences: path/to/new_genome_sequences.fasta <br>input_metadata: path/to/new_genome_sequences_metadata.csv &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; input_id_column: sequence_id<br><br>tree_annotations: pango_lineage,country<br><br>report_title: &quot;ARTIC Train Run 2022-08-05&quot; </span></h4><h4 class="c7" id="h.lx8axe47t528"><span class="c1">query_table_content: sample,barcode,country,collection_date,catchment</span></h4><h4 class="c7 c5" id="h.9moizuk2e4ro"><span class="c1"></span></h4><h4 class="c7" id="h.rufqwi43bf37"><span class="c1">report_preset: the_usual<br>input_date_column: collection_date </span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">The additions here change the title of the report, change the annotations on the tree tips, and specify that we want to run &ldquo;the usual&rdquo; analysis, which will produce a report containing summary tables, the catchment phylogenies, snipit plots and a timeline of samples. Because it now specifies a plot that includes a timeline, we need to tell civet which column to parse date information from (by default it will use `sample_date` but if our metadata has anything other than that we&rsquo;ll need to specify what that column name is. In this case we&rsquo;ve told it to look for data in the column `collection_date`. </span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><hr><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Civet has many, many configurable options. To see the full command-line usage of civet pull up the help menu:</span></p><h4 class="c7" id="h.ogt5roq6cjox"><span class="c1">(civet) artic$ civet --help</span></h4><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span class="c0">Also check out the full documentation for civet at cov-lineages.org/resources/civet.html</span></p><p class="c3 c5"><span class="c0"></span></p><p class="c3"><span>If anything isn&rsquo;t clear, or civet does something funky you&rsquo;re not expecting, feel free to post an issue to the Github repository for civet </span><span class="c10"><a class="c12" href="https://www.google.com/url?q=https://github.com/artic-network/civet&amp;sa=D&amp;source=editors&amp;ust=1662644105971016&amp;usg=AOvVaw3UiBrntbYu2QREAYzSUe0N">here</a></span><span>.</span></p><p class="c3 c5"><span class="c0"></span></p></body></html>